src/frontends/lean/decl_cmds.cpp

```diff
@@ -559,14 +559,28 @@ static environment modifiers_cmd(parser & p, ast_id & cmd_id, cmd_meta const & _
         mods.push(tk);
         tk = 0;
         p.next();
+        bool force_noncomputable = false;
+        if (p.curr_is_token_or_id(get_exclam_tk())) {
+            p.new_ast(get_exclam_tk(), p.pos());
```

Comment:
This should be reflected in the AST.
cc @digama0

Reply:
Yes, this is not correct since it does not store the generated AST node anywhere. I think it would be best to just pretend that `noncomputable!` is a single token here: if the `!` token is parsed after `noncomputable` then it creates a `noncomputable!` token in the AST, otherwise it just creates a `noncomputable` token. (This then has to be handled on the mathport side.)